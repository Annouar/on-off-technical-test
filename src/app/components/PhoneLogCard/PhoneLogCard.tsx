import React, { FC } from 'react';

import { PhoneLog } from 'types/PhoneLogs';
import { getTwoDigitsTime } from 'app/utils/date';

import {
  Card,
  CallInformationContainer,
  CallNumber,
  PhoneType,
  Time,
} from './styled';
import { Avatar } from '../Avatar/Avatar';

interface Props {
  phoneNumber: PhoneLog['phoneNumber'];
  phoneType: PhoneLog['phoneType'];
  date: PhoneLog['date'];
}

export const PhoneLogCard: FC<Props> = ({ phoneNumber, phoneType, date }) => {
  const phoneLogDate = new Date(date);

  const hours = getTwoDigitsTime(phoneLogDate.getUTCHours());
  const minutes = getTwoDigitsTime(phoneLogDate.getUTCMinutes());

  return (
    <Card>
      <Avatar />
      <CallInformationContainer>
        <CallNumber>+{phoneNumber}</CallNumber>
        <PhoneType>{phoneType}</PhoneType>
      </CallInformationContainer>
      <Time>
        {hours}:{minutes}
      </Time>
    </Card>
  );
};
