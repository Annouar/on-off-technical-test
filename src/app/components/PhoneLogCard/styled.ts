import styled from 'styled-components/macro';

export const Card = styled.div`
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 20px;
  background-color: white;
  border: 1px solid;
  border-radius: 20px;

  :hover {
    border: 2px solid green;
  }
`;

export const CallInformationContainer = styled.div`
  flex-basis: 60%;
  display: flex;
  flex-flow: column nowrap;
`;

export const CallNumber = styled.span``;

export const PhoneType = styled.span`
  color: #a9aaac;
`;

export const Time = styled.span`
  color: #a7a9aa;
`;
