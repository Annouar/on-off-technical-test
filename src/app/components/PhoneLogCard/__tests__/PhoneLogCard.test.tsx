import * as React from 'react';
import { render } from '@testing-library/react';

import { PhoneLogCard } from '../PhoneLogCard';

describe('<CallCard  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(
      <PhoneLogCard
        phoneNumber="33629210017"
        phoneType="Mobile"
        date="2021-04-06T10:20:06+0000"
      />,
    );

    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
