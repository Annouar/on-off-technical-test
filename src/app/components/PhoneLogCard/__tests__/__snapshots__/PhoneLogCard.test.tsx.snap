// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallCard  /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px;
  background-color: white;
  border: 1px solid;
  border-radius: 20px;
}

.c0:hover {
  border: 2px solid green;
}

.c3 {
  -webkit-flex-basis: 60%;
  -ms-flex-preferred-size: 60%;
  flex-basis: 60%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

.c4 {
  color: #a9aaac;
}

.c5 {
  color: #a7a9aa;
}

.c1 {
  background-color: #ccc;
  border-radius: 50%;
  height: 4rem;
  text-align: center;
  width: 4rem;
}

.c2 {
  font-size: 2rem;
  line-height: 1;
  position: relative;
  top: 1rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="c2"
    >
      #
    </span>
  </div>
  <div
    class="c3"
  >
    <span
      class=""
    >
      +
      33629210017
    </span>
    <span
      class="c4"
    >
      Mobile
    </span>
  </div>
  <span
    class="c5"
  >
    10
    :
    20
  </span>
</div>
`;
